src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -257,85 +260,132 @@ private JsonPropertyInfo GetPropertyWithUniqueAttribute(Type attributeType, Dict
             return property;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryIsPropertyRefEqual(in PropertyRef propertyRef, ReadOnlySpan<byte> propertyName, ulong key, ref JsonPropertyInfo info)
```

Comment:
Let's keep unrelated refactoring/moves like this method move separate from this PR (primarily because potential ports to 3.1 will be easier if the diffs are smaller/targeted).